File: be-smart-job/src/main/java/be_smart_job/service/social/interfaces/FavoriteService.java
package be_smart_job.service.social.interfaces;

import be_smart_job.dto.res.social.FavoriteFreelancerResponse;
import be_smart_job.dto.res.social.FavoriteJobResponse;

import java.util.List;

public interface FavoriteService {
    // Thao tác của Freelancer với Job
    boolean toggleFavoriteJob(String jobId);
    List<FavoriteJobResponse> getMyFavoriteJobs();

    // Thao tác của Client với Freelancer
    boolean toggleFavoriteFreelancer(String freelancerId);
    List<FavoriteFreelancerResponse> getMyFavoriteFreelancers();
}

File: fe-smart-job/src/services/favoriteService.js
import axiosClient from '../config/axiosClient';

export const favoriteService = {
  // Dành cho Freelancer
  toggleFavoriteJob: (jobId) => axiosClient.post(`/favorites/jobs/${jobId}`),
  getMyFavoriteJobs: () => axiosClient.get('/favorites/jobs'),

  // Dành cho Client
  toggleFavoriteFreelancer: (freelancerUserId) => 
    axiosClient.post(`/favorites/freelancers/${freelancerUserId}`),
  getMyFavoriteFreelancers: () => axiosClient.get('/favorites/freelancers'),
};